std/string.d

```diff
@@ -3857,37 +3857,52 @@ unittest
     Returns:
         GC allocated string with tabs replaced with spaces
   +/
-S detab(S)(S s, size_t tabSize = 8) pure
-    if (isSomeString!S)
+auto detab(Range)(auto ref Range s, size_t tabSize = 8) pure
+    if ((isForwardRange!Range && isSomeChar!(ElementEncodingType!Range))
+        || __traits(compiles, StringTypeOf!Range))
+{
+    import std.array;
+    return detabber(cast(StringTypeOf!Range)s, tabSize).array;
```

Comment:
I suspect this cast doesn't work for ranges that are not arrays. Just forwarding s should be good enough.